metricflow/data_table/mf_table.py

```diff
@@ -0,0 +1,279 @@
+from __future__ import annotations
+
+import datetime
+import itertools
+import logging
+from dataclasses import dataclass
+from decimal import Decimal
+from typing import Iterable, Iterator, List, Optional, Sequence, Tuple, Type
+
+import tabulate
+from metricflow_semantics.mf_logging.formatting import indent
+from metricflow_semantics.mf_logging.pretty_print import mf_pformat, mf_pformat_many
+from typing_extensions import Self
+
+from metricflow.data_table.column_types import CellValue, InputCellValue, row_cell_types
+from metricflow.data_table.mf_column import ColumnDescription
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True, eq=False)
+class MetricFlowDataTable:
```

Comment:
This is way more than I was expecting. I was thinking we could define a protocol that would let us implement wrappers around Agate/Pandas interfaces, cut over, and remove Pandas that way, not that we'd have a full bore data container class, but if this is what it takes let's just proceed.

Reply:
I thought this would be small going in, but the type-handling threw a wrench in there. We could still move to a protocol and this can help to come up with that definition since it has a limited interface.